<a href="https://colab.research.google.com/github/zack-murray/DS-Unit-1-Sprint-2-Statistics/blob/master/module1/LS_DS14_121_Statistics_Probability_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img align="left" src="https://lever-client-logos.s3.amazonaws.com/864372b1-534c-480e-acd5-9711f850815c-1524247202159.png" width=200>
<br></br>
<br></br>

## *Data Science Unit 1 Sprint 2 Assignment 1*

# Apply the t-test to real data

Your assignment is to determine which issues have "statistically significant" differences between political parties in this [1980s congressional voting data](https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records). The data consists of 435 instances (one for each congressperson), a class (democrat or republican), and 16 binary attributes (yes or no for voting for or against certain issues). Be aware - there are missing values!

Your goals:

1. Load and clean the data (or determine the best method to drop observations when running tests)
2. Using hypothesis testing, find an issue that democrats support more than republicans with p < 0.01
3. Using hypothesis testing, find an issue that republicans support more than democrats with p < 0.01
4. Using hypothesis testing, find an issue where the difference between republicans and democrats has p > 0.1 (i.e. there may not be much of a difference)

Note that this data will involve *2 sample* t-tests, because you're comparing averages across two groups (republicans and democrats) rather than a single group against a null hypothesis.

In [0]:
!wget https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data

In [0]:
import pandas as pd
import numpy as np 
import math
from scipy import stats

df = pd.read_csv('house-votes-84.data', names=['party','handicapped-infants','water-project',
                          'budget','physician-fee-freeze', 'el-salvador-aid',
                          'religious-groups','anti-satellite-ban',
                          'aid-to-contras','mx-missile','immigration',
                          'synfuels', 'education', 'right-to-sue','crime','duty-free',
                          'south-africa'])

df = df.replace({'y': 1, 'n': 0, '?': np.NaN})

rep = df[df['party']=='republican']
dem = df[df['party']=='democrat']



---

*Null Hypothesis*:

The two samples are the same


*Alternative Hypothesis*: 

The two samples are different


*Confidence Level*: 99%

---



In [0]:
stats.ttest_ind(dem['duty-free'], rep['duty-free'], nan_policy='omit')

Ttest_indResult(statistic=12.853146132542978, pvalue=5.997697174347365e-32)

In [0]:
rep['duty-free'].mean()

0.08974358974358974

In [0]:
dem['duty-free'].mean()

0.6374501992031872



---

*Conclusion*: Based on a p-value ~0, I reject the null hypothesis, that democrats and republicans support this bill at the same level, in same proportions. 

---



---

*Null Hypothesis*:

The two samples are the same


*Alternative Hypothesis*: 

The two samples are different


*Confidence Level*: 99%

---


In [0]:
stats.ttest_ind(dem['physician-fee-freeze'], rep['physician-fee-freeze'], nan_policy='omit')

Ttest_indResult(statistic=-49.36708157301406, pvalue=1.994262314074344e-177)

In [0]:
rep['physician-fee-freeze'].mean()

0.9878787878787879

In [0]:
dem['physician-fee-freeze'].mean()

0.05405405405405406

---

*Conclusion*: Based on a p-value ~0, I reject the null hypothesis, that democrats and republicans support this bill at the same level, in same proportions. 

---



---

*Null Hypothesis*:

The two samples are the same



*Alternative Hypothesis*: 

The two samples are different



*Confidence Level*: 90%

---



In [0]:
stats.ttest_ind(dem['water-project'], rep['water-project'], nan_policy='omit')

Ttest_indResult(statistic=-0.08896538137868286, pvalue=0.9291556823993485)

In [0]:
rep['water-project'].mean()

0.5067567567567568

In [0]:
dem['water-project'].mean()

0.502092050209205

---

*Conclusion*: Since the p-value of .929 is greater than it's confidence level of .1, I fail to reject the null hypothesis. The democrats and republicans support this bill at the roughly the same level, in same proportions. 

---

## Stretch Goals:

1. Refactor your code into functions so it's easy to rerun with arbitrary variables
2. Work on Performing a T-test without using Scipy in order to get "under the hood" and learn more thoroughly about this topic.
### Start with a 1-sample t-test
 - Establish the conditions for your test 
 - [Calculate the T Statistic](https://blog.minitab.com/hs-fs/hubfs/Imported_Blog_Media/701f9c0efa98a38fb397f3c3ec459b66.png?width=247&height=172&name=701f9c0efa98a38fb397f3c3ec459b66.png) (You'll need to omit NaN values from your sample).
 - Translate that t-statistic into a P-value. You can use a [table](https://www.google.com/search?q=t+statistic+table) or the [University of Iowa Applet](https://homepage.divms.uiowa.edu/~mbognar/applets/t.html)

 ### Then try a 2-sample t-test
 - Establish the conditions for your test 
 - [Calculate the T Statistic](https://lh3.googleusercontent.com/proxy/rJJ5ZOL9ZDvKOOeBihXoZDgfk7uv1YsRzSQ1Tc10RX-r2HrRpRLVqlE9CWX23csYQXcTniFwlBg3H-qR8MKJPBGnjwndqlhDX3JxoDE5Yg) (You'll need to omit NaN values from your sample).
 - Translate that t-statistic into a P-value. You can use a [table](https://www.google.com/search?q=t+statistic+table) or the [University of Iowa Applet](https://homepage.divms.uiowa.edu/~mbognar/applets/t.html)

 ### Then check your Answers using Scipy!



---

Refactor your code into functions so it's easy to rerun with arbitrary variables

---



In [0]:
def TST(a, b, nan_policy='omit'):
  return stats.ttest_ind(a, b, nan_policy='omit')

TST(rep['duty-free'], dem['duty-free'])

Ttest_indResult(statistic=-12.853146132542978, pvalue=5.997697174347365e-32)

In [0]:
TST(rep['water-project'], dem['water-project'])

Ttest_indResult(statistic=0.08896538137868286, pvalue=0.9291556823993485)



---

### Start with a 1-sample t-test
 - Establish the conditions for your test 
 - [Calculate the T Statistic](https://blog.minitab.com/hs-fs/hubfs/Imported_Blog_Media/701f9c0efa98a38fb397f3c3ec459b66.png?width=247&height=172&name=701f9c0efa98a38fb397f3c3ec459b66.png) (You'll need to omit NaN values from your sample).
 - Translate that t-statistic into a P-value. You can use a [table](https://www.google.com/search?q=t+statistic+table) or the [University of Iowa Applet](https://homepage.divms.uiowa.edu/~mbognar/applets/t.html)

---



---
Democrats on the mx-missle bill 

*Null Hypothesis*:

Republicans are very against this issue, only 25% support the bill.

*Alternative Hypothesis*: 

Republican support is not 25%.

*Confidence Level*: 95%

---

In [0]:
rep.shape

(168, 17)

In [0]:
rep['mx-missile'].mean()

0.11515151515151516

In [0]:
pd.Series(rep['mx-missile']).describe()

count    165.000000
mean       0.115152
std        0.320176
min        0.000000
25%        0.000000
50%        0.000000
75%        0.000000
max        1.000000
Name: mx-missile, dtype: float64

In [0]:
math.sqrt(168)

12.96148139681572

In [0]:
t = (.12 - .25)/(.32/12.96)

t

-5.265000000000001

In [0]:
def OST(a, b, nan_policy='omit'):
  return stats.ttest_1samp(a, b, nan_policy='omit')

OST(rep['mx-missile'], .25)

Ttest_1sampResult(statistic=-5.410018011156544, pvalue=2.1986730063257664e-07)

In [0]:
dem['mx-missile'].mean()

0.7580645161290323



---

*Conclusion*: Based on a p-value ~0, I reject the null hypothesis, that only 25 percent of republicans support the bill. 


---


